roles/usegalaxy-eu.find-miners/templates/find_miners.py

```diff
@@ -0,0 +1,521 @@
+#!/usr/bin/env python
+# A command line script that iterates over the currently running jobs and stops them as well as logs the user,
+# when a file in the JWD matches to a list of hashes
+
+import argparse
+import datetime
+import hashlib
+import os
+import pathlib
+import sys
+import time
+import zlib
+
+import galaxy_jwd
+import yaml
+from tqdm import tqdm
+
+CHECKSUM_FILE_ENV = "MALWARE_LIB"
+
+CURRENT_TIME = int(time.time())
+
+
+def convert_arg_to_byte(mb: str) -> int:
+    return int(mb) << 20
+
+
+def convert_arg_to_seconds(hours: str) -> int:
+    return int(hours) * 60 * 60
+
+
+def make_parser() -> argparse.ArgumentParser:
+    my_parser = argparse.ArgumentParser(
+        prog="Miner Finder",
+        description="""
+            Loads a yaml malware library with CRC32 and SHA1 hashes as arguments
+            from the environment variable "MALWARE_LIB",
+            searches in JWDs of currently running jobs for matching files
+            and reports jobs, users and malware details if specified.
+            Malware library file has the following schema:
+                class:
+                    name:
+                        version:
+                            severity: [high, medium, low]
+                            description: "optional info"
+                            checksums:
+                                crc32: <checksum crc32, gzip algorithm, integer representation>
+                                sha1: <checksum sha1, hex representation>
+            WARNING:
+            Be careful with how you generate the CRC32 hashes:
+            There are multiple algorithms, this script is using the one specified by IEEE 802.3
+            You should get this when using the gzip command on POSIX systems
```

Comment:
Do you have an example CMD here? I guess we want to make contributions to this library easy.

Reply:
yes I should add it. my current solution is only in the yaml file and reads not very smoothly.
`gzip -1 -c /path/to/file | tail -c8 | hexdump -n4 -e '"%u"'`